web: increase typescript use for ducks

mhils opened the pull request:
#### Description

In preparation for https://github.com/mitmproxy/mitmproxy/pull/7617, this PR adds full typescript typing for all redux actions emitted server-side.

Commit of mitmproxy/tools/web/static/app.js is intentional as backend message format changed; otherwise mitmweb is broken on main.

#### Checklist

 - [x] I have updated tests where applicable.

lups2000 commented:
Wow, amazing work! 🚀 This improves type safety even more. The code looks great to me; you just missed a couple of leftover imports. I haven’t tested it yet, but I’ll do that tomorrow to make sure everything is working fine together :)

lups2000 commented:
I tested it and everything seems to be working fine :) Once this is merged, I’ll merge the updates into **search-highlight-body-content** and then update #7617 by resolving the conflicts. What do you think?

mhils commented:
Added more tests and moved eventLog off utils/store. Please re-review. :)

The only remaining thing on utils/store is flow. This means we can integrate that into flows as part of the body filtering PR. :)

lups2000 commented:
> The only remaining thing on utils/store is flow. This means we can integrate that into flows as part of the body filtering PR. 

Not sure that I have understood, could you please expand on that? Thanks :)

lups2000 reviewed (approved):
This looks amazing now! Thanks for improving the test coverage 😄
